doc/guides/element_value_types.rst

```diff
@@ -0,0 +1,123 @@
+============================
+Element VRs and Python types
+============================
+
+.. currentmodule:: pydicom
+
+DICOM elements can contain anything from ASCII strings to unicode text,
+decimals, floats, signed and unsigned integers of different byte-depth and
+even encoded data. The format of the value of an element is given by its
+:dcm:`Value Representation<part05/sect_6.2.html>` or VR, and a list of VRs is
+given in the DICOM Standard in Part 5,
+:dcm:`Table 6.2-1 <part05/sect_6.2.html#table_6.2-1>`.
+
+So when using *pydicom*, what Python type should be used with a given VR to
+ensure that the value gets written correctly?
+
+* Elements of any VR:
+
+  * Can be set as empty by using ``None``
+  * Can have their values set using their *set using* or *stored as* type from
+    the table below
+
+* Non-**SQ** element values:
+
+  * Can also be set using a :class:`list` of their *set using* type - for
+    Value Multiplicity (VM) > 1, the value will be stored as a
+    :class:`~multival.MultiValue` of their *stored as* type
+
+* **SQ** element values should be set using a :class:`list` of zero or more
+  :class:`~dataset.Dataset` instances.
+* To ensure **AT** elements are encoded correctly, their values should be set
+  using either the 8-byte integer form of the tag - such as ``0x00100010`` for
+  (0010,0010) - or a :class:`~tag.BaseTag` and not as a 2-tuple or 2-list.
+
```

Comment:
Well, since mention here that can construct from a BaseTag, should it be an option in the table?  Not really a normal way to do it, but I did notice it missing when stated just above the table.

Also, makes me wonder whether we should fix it so that any valid tag form can be passed.

Reply:
It gets stored that way, and the table is more about using Python types to set values. There is a point that says you can use stored as values to set.

Reply:
It might make sense to just remove the reference to BaseTag. I want to change that example tag to `0x00100020` to make it less ambiguous anyway.

Reply:
Sure, sounds reasonable.

> I've been trying to think how to do it since you have to differentiate it from a tuple or list of 2 tags

Can't we just accept anything that doesn't throw an exception with Tag(arg)?

Reply:
Oh, right, multi-valued...

Reply:
How do you differentiate between:

```python
# List of two tags (0000,0000) and (0000,0001)
elem.value = [0x0000, 0x0001]
# One tag (0000,0001)
elem.value = [0x0000, 0x0001]